src/pages/index.js

```diff
@@ -1,67 +1,214 @@
+const api = new API({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b656ee04-ff76-4b23-b20a-1e4991fc7f99",
+    "Content-Type": "application/json",
+  },
+});
+
 // -------------------------------------------------------------------------------------
-//                              UserInfo Instance
+// UserInfo Instance
 // -------------------------------------------------------------------------------------
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
-  descriptionSelector: ".profile__description",
+  aboutSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
+
 // -------------------------------------------------------------------------------------
-//                                 Popup Instances
+// Inital UserInfo Population
 // -------------------------------------------------------------------------------------
-const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
-  handleFormSubmit: (formData) => {
+api
+  .getProfileData()
+  .then((data) => {
     userInfo.setUserInfo({
-      name: formData["name"],
-      description: formData["description"],
+      name: data.name,
+      about: data.about,
+      avatar: data.avatar,
     });
-    profileEditPopup.close();
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile data:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserInfo Popup
+// -------------------------------------------------------------------------------------
+const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
+  handleFormSubmit: (formData) => {
+    profileEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editProfileData(formData)
+      .then((data) => {
+        userInfo.setUserInfo(data);
+      })
+      .catch((error) => {
+        console.log("Error updating Profile:", error);
+      })
+      .finally(() => {
+        profileEditPopup.toggleUploadIndicator(false);
+        profileEditPopup.close();
+      });
   },
 });
+
 profileEditPopup.setEventListeners();
 
+// -------------------------------------------------------------------------------------
+// Editing UserAvatar Popup
+// -------------------------------------------------------------------------------------
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", {
+  handleFormSubmit: (formData) => {
+    const avatarURL = formData["Image URL"];
+    avatarEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editUserAvatar({ avatar: avatarURL })
+      .then((data) => {
+        userInfo.setUserInfo({
+          avatar: data.avatar,
+        });
+      })
+      .catch((error) => {
+        console.log("Error updating Avatar:", error);
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+        avatarEditPopup.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**